src/libs/device.js

```diff
@@ -1,10 +1,15 @@
-export const MOBILE_THRESHOLD = 640;
-
-export default class Device {
-  /**
-   * return if device is mobile according to the viewport width
-   */
-  static isMobile() {
-    return window.innerWidth < MOBILE_THRESHOLD;
-  }
+import React from 'react';
+
+const mediaQuery = window.matchMedia('(max-width: 640px)');
+
+export function isMobileDevice() {
+  return mediaQuery.matches;
 }
+
+export function onDeviceSizeChange(callback /* isMobile => {…} */ ) {
+  return mediaQuery.addListener(mediaQueryEvent => {
```

Comment:
We need to removeListener once the component is unmounted to avoid memory leak

Reply:
I voluntarily omitted it as the only case when PanelManager is unmounted is the app being exited. But maybe it's safer to do it anyway :man_shrugging:

Reply:
Did it. I didn't see the point of an abstraction anymore, so I exposed the media query itself.